src/types/objects.rs

```diff
@@ -0,0 +1,85 @@
+//! Implementations for the types describing DIF object files.
+
+use super::{
+    AllObjectCandidates, Arch, CodeId, CompleteObjectInfo, DebugId, ObjectCandidate,
+    ObjectFeatures, ObjectFileStatus, ObjectUseInfo, RawObjectInfo,
+};
+use crate::sources::{SourceId, SourceLocation};
+
+impl Default for ObjectUseInfo {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+impl ObjectUseInfo {
+    pub fn is_none(&self) -> bool {
+        matches!(self, Self::None)
+    }
+}
+
+impl From<Vec<ObjectCandidate>> for AllObjectCandidates {
```

Comment:
Maybe `IntoIterator<ObjectCandidate>` would be a better bound?

Reply:
This way this can move the Vec rather than re-create it.  And this happens to already be created as a Vec, so I don't think there's anything to be gained from being more generic here.

Maybe this should be a `::new` method intead?  I'm not really sure when new is preferred.